### ISSUE TITLE
When can we expect Tus 2.0 implementation for .NET?

### ISSUE THREAD

**Issue by goradata** [tusdotnet/tusdotnet#208]

**Comment by smatsson** [comment]
Hi! 

[Resumable Uploads For Http](https://datatracker.ietf.org/doc/draft-ietf-httpbis-resumable-upload/) (working name was "tus2") is new proposed standard by the IETF. The code related to this spec is located in [POC/tus2](https://github.com/tusdotnet/tusdotnet/tree/POC/tus2) and has been verified to work with the latest draft spec (v 01 as of this writing).

The nuget package won't contain this code until the draft becomes an RFC and the timeline for that is unclear. You can follow the work over at the [IETF's github page](https://github.com/httpwg/http-extensions/).

The code in the POC/tus2 branch has been verified to work with the clients that are currently available so it should be ready to use by compiling it and including it in your projects. See clients and compatibility list [here](https://github.com/tus/draft-example).

**Comment by goradata** [comment]
According to [Marius's post](https://tus.io/blog/2023/09/20/tusd-200) there is a seperate Tus 2.0 release (in go)  where IETF status kept as draft which has lots of new features. From the dates on POC/tus2 looks like it is eariler than this announcement. Does this include all the improvements that post is talking about?

**Comment by smatsson** [comment]
I can see why this is confusing. :) The linked blog post announces the new tusd 2.0 release (notice the last `d`). tusd is the official server implementation for go. Both tusd and tusdotnet implements the same version of Resumable Uploads For HTTP (the latest one, which is currently v01).

Hope this clears it up.

**Comment by goradata** [comment]
Yes it does, thanks for clarification. From the blog post I understand tusd 2.0 has reliability and performance improvements (regardless of the tus protocol version). I am assuming that there might be some areas of improvements for tusdotnet based on the updated go implementation. Are you planning on carrying over any improvements?

**Comment by smatsson** [comment]
Great! Most of the improvements doesn't really apply to the dotnet version as these are specific to the go implementation. tusd and tusdotnet live their own lives and are not really a 1:1 comparison. There are ongoing work in tusdotnet to improve performance and resilience (see e.g. #205 ). I think the idea of backporting the locking behavior from `Resumable Uploads For Http` is interesting and something worth investigating more as it's already implemented in the POC/tus2 branch. I know that this would break some use cases in tusdotnet so it would probably need to be considered an off switch for it or introduce a breaking change version, which I'm not keen on.

**Comment by loligans** [comment]
I was reviewing the IETF draft and it looks pretty close to ratification. Are there any structural protocol changes that have been made since the POC branch's current draft implementation? (I believe the POC has implemented revisison 6, and we're on 11 or 12 now)
